v2/tools/generator/internal/config/type_configuration.go

```diff
@@ -27,9 +27,10 @@ import (
 //
 type TypeConfiguration struct {
 	name              string
-	renamedTo         *string
-	renamedToConsumed bool
 	properties        map[string]*PropertyConfiguration
+	nameInNextVersion configurableString
```

Comment:
The configurableX types are nice - I guess they'll be even nicer once we can use generics!